Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Planner-stage gives us `recipe.json` which we use in Cacher-stage
+FROM rust:1.54.0-buster as planner
+
+WORKDIR /app
+
+RUN curl https://getsubstrate.io -sSf | bash -s \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+COPY . .
+RUN cargo build --release \
+    && mv target/release/node-template /usr/local/bin/supra-dht \
+    && mv /usr/local/cargo/bin/sub* /usr/local/bin/ \
+    && rm -rf target \
+    && rm -rf /root/.cargo/registry/ \
+    && rm -rf /tmp && rm -rf /usr/local/rustup && rm -rf /usr/local/cargo
+
+FROM rust:1.54.0-buster as runtime
+COPY --from=planner /usr/local/bin/supra-dht /usr/local/bin/
+COPY --from=planner /usr/local/bin/sub* /usr/local/bin/
+
```

Comment:
The goal is to make this image reusable so that we can launch new nodes on other devices by just providing custom keys, ports and the rest. From the part won't be necessary. If possible in Docker, let it expect the arguments in CMD when running the image locally

Reply:
Arguments passed in `CMD` will be overridden when we provide our own.
For example, when we want a non-authorize node to be started, we can do this:

```bash
docker run -it -p 30333:30333 -p 9944:9944 -p 9933:9933 supra-dht \
  --base-path /tmp/bob \
  --chain local \
  --bob \
  --port 30333 \
  --ws-port 9944 \
  --rpc-port 9933 \
  --no-telemetry \
  --no-prometheus \
  --validator \
  --bootnodes /ip4/127.0.0.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp  
```

With `docker run -p` we can map the ports exposed by image to any port on the host.